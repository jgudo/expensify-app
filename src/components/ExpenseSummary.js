import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import getExpenseTotal from '../selectors/expenses-total';
import visibleExpenses from '../selectors/expenses';
import numeral from 'numeral';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const ExpenseSummary = ({expenseCount, expenseTotal}) => {
    return (
        <div className="page-header">
           <div className="content-container">
                {expenseCount >=1 && (
                    <h2 className="page-header__title">Viewing 
                        <span> { expenseCount }</span> 
                        { expenseCount <= 1 ? ' expense ' : ' expenses '}
                        totalling 
                        <span>   ₱{numeral(expenseTotal / 100).format('₱0,0.00')}</span>
                    </h2>
                )}
                <div className="page-header__actions">
                    <Link to="/create" className="button button--action">
                         <span>Add Expense</span>
                         <span><FontAwesomeIcon icon={['fas', 'plus']} /> </span>
                    </Link>
                   
                </div>
           </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visible = visibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visible.length,
        expenseTotal: getExpenseTotal(visible)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);