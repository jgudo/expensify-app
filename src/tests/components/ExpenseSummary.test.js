import React from 'react';
import {shallow} from 'enzyme';
import ExpenseSummary from '../../components/ExpenseSummary';

test('Should render ExpenseSummary correctly', () => {
    const wrapper = shallow(<ExpenseSummary />);
    expect(wrapper).toMatchSnapshot();
});