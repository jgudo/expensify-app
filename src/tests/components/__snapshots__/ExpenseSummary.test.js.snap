// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
